Adapter_Revit_UI/Create/Create.cs

```diff
@@ -260,28 +295,55 @@ private static void SetIdentifiers(IBHoMObject bHoMObject, Element element)
             SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.ElementId, element.Id.IntegerValue);
             SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.AdapterId, element.UniqueId);
 
-            int aWorksetId = WorksetId.InvalidWorksetId.IntegerValue;
-            if (element.Document != null && element.Document.IsWorkshared)
+            if (element is Family)
             {
-                WorksetId aWorksetId_Revit = element.WorksetId;
-                if (aWorksetId_Revit != null)
-                    aWorksetId = aWorksetId_Revit.IntegerValue;
+                Family aFamily = (Family)element;
+
+                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyPlacementTypeName, Query.FamilyPlacementTypeName(aFamily));
+                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyName, aFamily.Name);
+                if (aFamily.FamilyCategory != null)
+                    SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.CategoryName, aFamily.FamilyCategory.Name);
             }
-            SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.WorksetId, aWorksetId);
+            else
+            {
+                int aWorksetId = WorksetId.InvalidWorksetId.IntegerValue;
+                if (element.Document != null && element.Document.IsWorkshared)
+                {
+                    WorksetId aWorksetId_Revit = element.WorksetId;
+                    if (aWorksetId_Revit != null)
+                        aWorksetId = aWorksetId_Revit.IntegerValue;
+                }
+                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.WorksetId, aWorksetId);
 
-            Parameter aParameter = null;
+                Parameter aParameter = null;
 
-            aParameter = element.get_Parameter(BuiltInParameter.ELEM_FAMILY_PARAM);
-            if (aParameter != null)
-                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyName, aParameter.AsValueString());
+                aParameter = element.get_Parameter(BuiltInParameter.ELEM_FAMILY_PARAM);
+                if (aParameter != null)
+                {
+                    string aValue = aParameter.AsValueString();
+                    if (string.IsNullOrEmpty(aValue))
```

Comment:
Check if this should be the opposite? Right now setting the values only if it is null or empty.